### PR TITLE
Add directive for nothing

### PR DESCRIPTION
**Pull request by janbrohl**
This directive doesn't load anything so can be used without any data source, eg for testing or nesting.

### CODE REVIEW

**Comment by dhellmann** [comment]
I like the idea, but I'm not sure about the name. How about datatemplate:nodata ? or nosource?

File: sphinxcontrib/datatemplates/directive.py [hunk 1]
```diff
@@ -136,6 +138,22 @@ class DataTemplateWithEncoding(DataTemplateBase):
                               })
 
 
+class DataTemplateNothing(DataTemplateBase):
+    """
+    .. rst:directive:: .. datatemplate:nothing:: source-path
```
Comment (dhellmann) [inline sphinxcontrib/datatemplates/directive.py]: If source-path isn't used, we shouldn't include it in the syntax.

**Comment by janbrohl** [comment]
I thought about datatemplate:none (as that is what it puts in data), datatemplate:ignore-source (as that is what ist effectively does), datatemplate:empty

File: sphinxcontrib/datatemplates/directive.py (hunk 1 above)
Comment (janbrohl) [reply sphinxcontrib/datatemplates/directive.py]: Actually source-path is available to the template as source (same for all directives) but it should be made clear.

**Comment by dhellmann** [comment]
> I thought about datatemplate:none (as that is what it puts in data), datatemplate:ignore-source (as that is what ist effectively does), datatemplate:empty

We want to name it from the perspective of the user, so they understand what it does rather than how it works. "empty" or "nodata" conveys the right meaning, I think.

File: sphinxcontrib/datatemplates/directive.py (hunk 1 above)
Comment (janbrohl) [reply sphinxcontrib/datatemplates/directive.py]: Removed. We should put something about argument and option handling in the docs though.

File: doc/source/index.rst [hunk 2]
```diff
@@ -14,13 +14,14 @@ for Sphinx to render as part of its output.
    :maxdepth: 2
 
    using
+   nothing
```
Comment (dhellmann) [inline doc/source/index.rst]:
```suggestion
   nodata
```

File: doc/source/nothing.rst [hunk 3]
```diff
@@ -0,0 +1,26 @@
+======================
+ Nothing Sample
```
Comment (dhellmann) [inline doc/source/nothing.rst]:
```suggestion
 No External Data Sample
```
(also rename the file)

**Review by dhellmann — CHANGES_REQUESTED** [review]

File: doc/source/nodata.rst [hunk 4]
```diff
@@ -0,0 +1,26 @@
+======================
+ No External Data Sample
+======================
```
Comment (dhellmann) [inline doc/source/nodata.rst]: The title lines here need to be longer to get the doc build to pass.

**Pull request merged**